### ISSUE TITLE
Add check option?

### ISSUE THREAD

**Issue by kaarposoft** [ufrisk/MemProcFS#275]
When running `memprocfs`, stdout may show eg.
`Could not load PDB required file - libpdbcrust.dll/so`
But despite this, `memprofs` continues, which is great "in general".

However, sometimes we (at least in my usecase) would need `memprocfs` with all the bells and whistles.
Thus I would propose a `memprofs` option -- maybe `-check`--  to return 0 if all prerequisites are found to be ok; -1 otherwise.

**Comment by kaarposoft** [comment]
Good point.

I do not know enough about the architecture and dependencies of MemProcFs to give a good answer.

But I can tell you my usecase:
My tool is dependent on several 3rd party tools.
Before anything else, I would like my tool to verify that those dependencies are available, working, and has the right version.
So for each dependency a launch a process which runs the tool with -V option or similar, check the return code, and grep for the version (see issue #274).
In the case of MemProcFs I then use the API to do whatever it needs to do to extract data.
The issue is that eg. the "libpdbcrust.so (symbol support)" message scrolls by on stdout, and it is easily missed. (I could grep for it of course, but what if there are other warnings tomorrow that I am not aware of).
So basically I would like the check option to ensure, that if I subsequently use the API it will work with "all the bells and whistles".

Hope this makes sense.

**Comment by ufrisk** [comment]
I'm going to close this specific issue without resolve. It would be very hard for me to implement since supporting DLLs are loaded at different stages during initialization; also other things may fail as well.

it's also very hard for me to know when to stop and when to not since that would be highly dependent on user context.

For now grepping the output or using the API is probably the way to go. If you feel anything is missing from the API (I would imagine some checks on various internal statuses might be missing) let me know and I'll do my best to add them.

I'll skip on the general -check option though due to complexity and ambiguity.
